src/Runner.Worker/JobExtension.cs

```diff
@@ -239,9 +239,23 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
 
                             actionRunner.TryEvaluateDisplayName(contextData, context);
                             jobSteps.Add(actionRunner);
+
```

Comment:
It looks like there is an assumption that main is always defined. Is there validation somewhere that enforces?

Reply:
i have the validation when i parse the action.yml file, it requires the node action to have the `main` and container action to have the `image`